python/lsst/afw/table/_syntax.py

```diff
@@ -202,3 +202,70 @@ def processArray(a):
         else:
             d[name] = processArray(self.get(schemaItem.key))
     return d
+
+def BaseCatalog_asAstropy(self, cls=None, copy=False, unviewable="copy"):
+    """!
+    Return an astropy.table.Table (or subclass thereof) view into this catalog.
+
+    @param[in]   cls        Table subclass to use; None implies astropy.table.Table itself.
+                            Use astropy.table.QTable to get Quantity columns.
+
+    @param[in]  copy        Whether to copy data from the LSST catalog to the astropy table.
+                            Not copying is usually faster, but can keep memory from being
+                            freed if columns are later removed from the Astropy view.
+
+    @param[in]  unviewable  One of the following options, indicating how to handle field types
```

Comment:
If there was no precedent here, I think I'd feel pretty strongly that `to_X()` methods are the better interface, because it puts the responsibility in the right place (if object A is viewable as a B, it's A that actually has to know how to do that), and it lets constructors stay simple.  And there's nothing stopping you from adding pandas constructor args to `to_pandas()`, and in fact you can limit it to just the pandas constructor arguments that make sense for this conversion. That's potentially an advantage over the constructor, which has to have all possible arguments for all possible construction patterns and thus pretty complex documentation to specify how those options interact.  You can use `**kwargs` to get around some of that, but I think it's always preferable to avoid that when you can.

All that said, using constructors for type casting is very much an established pattern in Python, even way down in the standard library.  I think that may have started largely because it's familiar to people coming from C++, Java, or other strongly-typed languages with overloading (where this pattern very much makes sense), but it's here now, and it's not something I can wish away.

I don't have a well-formed opinion about `from_X` vs. a constructor; generally I'd prefer the former to reduce overloading, but your point about replicating the full constructor API is a good one.

Anyhow, I think the current interface should accept `**kwargs` correctly in `__astropy_table__` already, so if `Table`'s constructor does allow those to be forwarded from the user, the type-casting constructor will indeed be as fully functional as the `asAstropy` method, and I'd be happy to eventually deprecate the latter if it ultimately proves unpopular in comparison.